app/src/main/java/com/mybus/service/ServiceFacade.java

```diff
@@ -0,0 +1,87 @@
+package com.mybus.service;
+
+import android.content.Context;
+
+import com.google.android.gms.maps.model.LatLng;
+import com.mybus.location.OnAddressGeocodingCompleteCallback;
+import com.mybus.location.OnLocationGeocodingCompleteCallback;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+
+public class ServiceFacade {
+
+    private static ServiceFacade instance = null;
+    private GisService gisService;
+    private MyBusService myBusService;
+    private GeocodingService geocodingService;
+
+    private ServiceFacade() {
+        gisService = new GisService();
+        myBusService = new MyBusService();
+        geocodingService = new GeocodingService();
+    }
```

Comment:
GisService, MyBusService and GeocodingService should be interfaces.
In the ServiceFacade's constructor we should specify the concrete classes implementing those interfaces.